### PR TITLE
fix(MastheadLeftNav): when user goes through menu levels, set scrollTop to 0

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Component: Mega Menu - scrolling issue on second level menu options #3730

### Description

When user scrolls in the mobile masthead menu, clicks on an option, the scroll position remains the same when user should be brought to the top of the next menu level.

This PR resets the scroll position every time the user clicks on an menu option

### Changelog

**Changed**

- set `bx--side-nav__items` scrollTop to 0 on click

### CODE REVIEW

File: packages/react/src/components/carbon-components-react/UIShell/SideNavMenu.js [hunk 1]
```diff
@@ -108,9 +108,21 @@ export class SideNavMenu extends React.Component {
     };
   }
 
+  scrollToTop = () => {
+    const navItemsElem = root.document.getElementsByClassName(
+      'bx--side-nav__items'
+    );
```
Comment (asudoh) [inline packages/react/src/components/carbon-components-react/UIShell/SideNavMenu.js:114]: It'll be nicer if we can better scope the selection of elements (e.g. only the menu items associated with this menu, or go up to the ancestor `.bx--side-nav` and start from there), but going as-is looks OK for now.

Comment (annawen1) [reply packages/react/src/components/carbon-components-react/UIShell/SideNavMenu.js]: hmm there's actually only one `bx--side-nav__items` element from `SideNavItems`, it wraps the entire menu. Would it be better to create a ref for the `SideNavMenu` and get the `bx--side-nav__items` from the ref instead?

Comment (jeffchew) [reply packages/react/src/components/carbon-components-react/UIShell/SideNavMenu.js]: It's certainly the "React" way of doing it...

Comment (annawen1) [reply packages/react/src/components/carbon-components-react/UIShell/SideNavMenu.js]: changed to target the `side-nav__items` element from the ref instead

**Review by jeffchew — APPROVED** [review]
LGTM!

**Pull request merged**